### PR TITLE
Add Apple M1 tests

### PR DESCRIPTION
**Pull request by anton-l**
This adds an M1-specific workflow to run fast test for `mps` devices.

### CODE REVIEW

**Comment by anton-l** [comment]
@pcuenca the tests are ready for inspection right in this PR :) 

Note: `[MPSTemporaryNDArray initWithDevice:descriptor:] Error: product of dimension sizes > 2**31'` outright segfaults, so we might not check all tests if the number of such failures is more than the number of pytest workers (4).

File: .github/workflows/pr_tests.yml [hunk 1]
```diff
@@ -10,14 +10,14 @@ concurrency:
   cancel-in-progress: true
 
 env:
-  HF_HOME: /mnt/cache
   OMP_NUM_THREADS: 8
   MKL_NUM_THREADS: 8
   PYTEST_TIMEOUT: 60
+  MPS_TORCH_VERSION: 1.13.0.dev20220830
```
Comment (anton-l) [inline .github/workflows/pr_tests.yml:16]: This is where we pin the torch version with the latest fixes for mps

**Comment by patrickvonplaten** [comment]
Looks very cool ! Feel free to merge whenever @anton-l - also fine if they fail for a bit on master and we fix step-by-step

**Comment by pcuenca** [comment]
Hi @anton-l !

I made a couple of changes:
- Use the RC of the upcoming PyTorch 1.13.
- Skip the test that crashed, added another one to test the attention layer that is compatible with MPS.
- Skip some cuda tests on MPS.

However, `AutoencoderKLTests.test_output_pretrained` is still failing, I don't know the reason why.

Could you please take a look at my changes? I think we can merge with this one failure and I'll try to fix it in a couple days.

File: tests/test_models_vae.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ def test_output_pretrained(self):
             model.config.in_channels,
             model.config.sample_size,
             model.config.sample_size,
-            generator=torch.manual_seed(0),
+            generator=generator,
```
Comment (pcuenca) [inline tests/test_models_vae.py:99]: I think this was the original intention. Unfortunately it doesn't fix the failing test.

**Review by pcuenca — APPROVED** [review]

File: tests/test_models_vae.py (hunk 2 above)
Comment (patrickvonplaten) [reply tests/test_models_vae.py:99]: @anton-l I think this is the reason why some tests are currently failing on master - could you take a look?

**Pull request merged**